torch_geometric/nn/aggr/basic.py

```diff
@@ -66,6 +66,130 @@ def forward(self, x: Tensor, index: Optional[Tensor] = None,
         return torch.sqrt(var.relu() + 1e-5)
 
 
+class QuantileAggregation(Aggregation):
+    r"""Quantile aggregation operator.
+
+    Args:
+        q (float): A scalar in the range [0, 1].
+        interpolation (str): Interpolation method applied if the quantile point
+            :math:`q(n - 1)` lies between two values :math:`x_i \le x_{i+1}`,
+            with :math:`i < q(n - 1) < i+1`. Can be one of the following:
+
+            - :obj:`"lower"`: Returns the one with lowest value.
+
+            - :obj:`"higher"`: Returns the one with highest value.
+
+            - :obj:`"midpoint"`: Returns the average of the two values.
+
+            - :obj:`"nearest"`: Returns the one nearest to the quantile point.
+
+            - :obj:`"linear"` (default): Returns a linear combination of the
+                two elments, as
+                :math:`x_i + (x_{i+1} - x_i)\cdot(q(n - 1) - i)`.
+
+        fill_value (float): Default value in the case no entry is
+            found for a given index (default: :obj:`NaN`).
+    """
+
+    interpolations = {'linear', 'lower', 'higher', 'nearest', 'midpoint'}
+
+    def __init__(self, q: float, interpolation: str = 'linear',
+                 fill_value: float = float('nan')):
+        if q > 1. or q < 0:
+            raise ValueError("'q' must be in the range [0, 1].")
+        if interpolation not in self.interpolations:
+            raise ValueError(f"Invalid '{interpolation}' interpolation method."
+                             f" Available interpolations:"
+                             f"{self.interpolations}")
+
+        super().__init__()
+        self.q = q
+        self.interpolation = interpolation
+        self.fill_value = fill_value
+
+    def forward(self, x: Tensor, index: Optional[Tensor] = None,
+                ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
+                dim: int = -2) -> Tensor:
+        # we could also expand `ptr` to an index
+        self.assert_index_present(index)
+
+        if dim_size is None:
+            dim_size = index.max() + 1
+
+        # compute the quantile points
+        count = torch.bincount(index, minlength=dim_size)
+        cumsum = torch.cumsum(count, dim=0)
+        q_point = self.q * (count - 1) + cumsum - count
+
+        # shape used for expansions
+        # (1, ..., 1, -1, 1, ..., 1)
+        shape = [1] * x.dim()
+        shape[dim] = -1
+
+        # two sorts: the first one on the value,
+        # the second (stable) on the indices
+        x, x_perm = torch.sort(x, dim=dim)
+        index = index.view(*shape).expand_as(x)
+        index = index.take_along_dim(x_perm, dim=dim)
+        index, index_perm = torch.sort(index, dim=dim, stable=True)
+        x = x.take_along_dim(index_perm, dim=dim)
+
+        # compute the quantile interpolations
+        if self.interpolation == 'lower':
+            idx = torch.floor(q_point).long()
+            median = x.index_select(dim, idx)
```

Comment:
```suggestion
            q = x.index_select(dim, idx)
```
or `quantile`?

Reply:
You are right, I started implementing `median` and then swithced to `quantile`, I forgot to change the naming